pumpkin/src/lan_broadcast.rs

```diff
@@ -0,0 +1,39 @@
+use pumpkin_config::ADVANCED_CONFIG;
+use std::net::{IpAddr, Ipv4Addr, SocketAddr};
+use std::time::Duration;
+use tokio::net::UdpSocket;
+use tokio::time;
+
+const BROADCAST_ADDRESS: SocketAddr =
+    SocketAddr::new(IpAddr::V4(Ipv4Addr::new(224, 0, 2, 60)), 4445);
+
+pub async fn start_lan_broadcast(bound_addr: SocketAddr) {
+    let port = ADVANCED_CONFIG.lan_broadcast.port.unwrap_or(0);
+
+    let socket = UdpSocket::bind(SocketAddr::new(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)), port))
+        .await
+        .expect("Unable to bind to address");
+
+    socket.set_broadcast(true).unwrap();
+
+    let mut interval = time::interval(Duration::from_millis(1500));
+    let advertisement = format!(
+        "[MOTD]{}[/MOTD][AD]{}[/AD]",
+        &ADVANCED_CONFIG.lan_broadcast.motd,
+        bound_addr.port()
+    );
+
+    log::info!(
+        "LAN broadcast running on {}",
+        socket
+            .local_addr()
+            .expect("Unable to find running address!")
+    );
```

Comment:
```suggestion
```
I don't think printing this info is necessary at all.

Reply:
The query does the same
https://github.com/Snowiiii/Pumpkin/blob/00af2b98683b11946cddb4f08de3b0ce04c1fddd/pumpkin/src/query.rs#L43-L48
I think this should be kept because the default port is set to 0 for LAN broadcast, this means the system will give you some open port and printing this assigned port sounds like a good Ida to me.

Reply:
But the thing is query requires clients to send packets to that specific port and address, but LAN broadcast doesn't require clients to send packets to whatever port and address they are bound to at all. Thats why I feel like printing this is unnecessary.

Reply:
It's still an open port, IDK. I would rather keep it and remove it when it is decided that It's too much log clutter.